papers/Repeaters/evaluate_mc_calculation.py

```diff
@@ -0,0 +1,522 @@
+""" 
+This script generates simulated instances of MC FRBs in z DM space for
+a single point in Rparameter space, and evaluates the likelihood over
+the rest of that space.
+
+It will first generate a file called "mc.npy", which contains
+the *indices* of repeaters in zDM space. The file structure is:
+- list over Nbin declination bins
+- list over NMC MC calculations (default: 100)
+- either empty (no FRBs generated) OR list of 3 indices: iz,iDM,NFRB
+- Each entry says "NFRB repeating FRBs were generated at iz,iDM"
+
+The second step is to evaluate the likelihood over the entire space.
+It first loops over the Rgamma,Rmax grid, loading in Rmin values from
+the input file generated by "fit_repeating_distributions.py".
+In each case, it calculates the likelihoods for repeating FRBs only,
+by first normalising the repeating zDM grid to unity, and then
+storing the sum of the log-likelihoods for each set of FRB
+observations. These get summed over all bins, but separate values
+are stored for each Nmc calculation. The outputs are kept in
+the "savefile".
+
+The third step is to plot the result. It loops over all NMC iterations,
+and for each, calculates a Baysian confidence intervals at 1,2,3 sigma
+limits. For each grid point, it then records whether or not it lies within
+that interval. Finally, it plots contours for all points that lie 50%
+of the time in/our of that interval.
+
+
+"""
+import os
+from pkg_resources import resource_filename
+from zdm import cosmology as cos
+from zdm import misc_functions
+from zdm import parameters
+from zdm import survey
+from zdm import pcosmic
+from zdm import iteration as it
+from zdm.craco import loading
+from zdm import io
+from zdm import repeat_grid as rep
+
+import utilities as ute
+import states as st
+
+import pickle
+import numpy as np
+from zdm import survey
+from matplotlib import pyplot as plt
+
+import scipy as sp
+from scipy.stats import poisson
+
+import states
+
+import matplotlib
+import time
+from zdm import beams
+
+
+beams.beams_path = '/Users/cjames/CRAFT/FRB_library/Git/H0paper/papers/Repeaters/BeamData/'
```

Comment:
are these the same files as in `zdm/papers/Repeaters/BeamData/` ?

Reply:
Yes, and this is an old pathname, now deprecated and removed. Files now shifted to zdm/data/BeamData/CHIME/